### PR TITLE
chore: Use alejandra for Nix formatting

### PR DESCRIPTION
**Pull request by ghost**
Also replaces lint-staged with pre-commit.nix

### CODE REVIEW

**Comment by louib-bitgo** [comment]
@ericcrosson-bitgo I've just realized that this will remove the pre-commit setup for devs not using Nix. Are we ok with that?

**Comment by ericcrosson-bitgo** [comment]
> @ericcrosson-bitgo I've just realized that this will remove the pre-commit setup for devs not using Nix. Are we ok with that?

That's a great point... I think it's fair to assume this would be a net negative for the average potential contributor. Not that we're flush with those yet, but a team can dream!

Do you think you can wire Nix in to a pre-commit hook using lint-staged?

**Comment by louib-bitgo** [comment]
@ericcrosson-bitgo 

> Do you think you can wire Nix in to a pre-commit hook using lint-staged?

You mean calling a Nix pre-commit hook from lint-stage? I guess I could, but that would still require Nix to be installed/activated, no? And then `eslint` would be called twice.

> That's a great point... I think it's fair to assume this would be a net negative for the average potential contributor. Not that we're flush with those yet, but a team can dream!

We could document that commit hooks are only available when using Nix

**Comment by ericcrosson-bitgo** [comment]
> You mean calling a Nix pre-commit hook from lint-stage? I guess I could, but that would still require Nix to be installed/activated, no? And then `eslint` would be called twice.

@louib-bitgo err no that was my unclear phrasing, I mean invoking alejandra from lint-staged. I guess having alejandra installed would be the real trick... :thinking: 

> We could document that commit hooks are only available when using Nix

This is true, though it still seems a subpar developer experience. I'm more tempted to apply alejandra once manually and call it a day, especially since it's only a formatter and doesn't provide static analysis

File: flake.nix [hunk 1]
```diff
@@ -1,21 +1,39 @@
+# Get started by following https://github.com/BitGo/bitgo-microservices/blob/develop/docs/NIX.md
 {
   description = "api-ts";
 
   inputs = {
     nixpkgs.url = "github:nixos/nixpkgs";
     flake-utils.url = "github:numtide/flake-utils";
+    pre-commit-hooks.url = "github:cachix/pre-commit-hooks.nix";
```
Comment (bitgopatmcl) [inline flake.nix:8]:
```suggestion
    pre-commit-hooks = {
      url = "github:cachix/pre-commit-hooks.nix";
      inputs.nixpkgs.follows = "nixpkgs";
      inputs.flake-utils.follows = "flake-utils";
    };
```

Overriding these transitive dependencies should remove the duplicate nixpkgs and flake-utils from the lockfile and improve performance

**Comment by ericcrosson-bitgo** [comment]
Requesting a temperature check on

> This is true, though it still seems a subpar developer experience. I'm more tempted to apply alejandra once manually and call it a day, especially since it's only a formatter and doesn't provide static analysis

Do you agree?

**Comment by louib-bitgo** [comment]
> This is true, though it still seems a subpar developer experience. I'm more tempted to apply alejandra once manually and call it a day, especially since it's only a formatter and doesn't provide static analysis

Yeah that makes sense. I'll do that and revert the other changes

File: flake.nix (hunk 1 above)
Comment (louib-bitgo) [reply flake.nix]: I reverted the pre-commit hook changes from this PR

**Review by ericcrosson-bitgo — APPROVED** [review]

**Pull request merged**